conf/modules.config

```
@@ -1,32 +1,332 @@
+        ext.args = { params.prodigal_trainingfile ? "-t $params.prodigal_trainingfile" : "-p meta" }
+        publishDir = [
+            path: { "${params.outdir}/prodigal" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: FORMAT_PRODIGAL_GFF {
+        publishDir = [
+            path: { "${params.outdir}/prodigal" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: PREDICT {
+        publishDir = [
+            path: { "${params.outdir}/transdecoder" },
+            mode: params.publish_dir_mode,
+            pattern: "*.transdecoder.*"
+        ]
+    }
+
+    withName: 'PROKKA' {
+        ext.args = '--prodigal --metagenome'
+    }
+
+    withName: 'FAA_CAT' {
+        ext.prefix = 'prokka.faa.gz'
+        publishDir = [
+            path: { "${params.outdir}/prokka" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: 'FFN_CAT' {
+        ext.prefix = 'prokka.ffn.gz'
+        publishDir = [
+            path: { "${params.outdir}/prokka" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: 'GFF_CAT' {
+        ext.prefix = 'prokka.gff.gz'
+        publishDir = [
+            path: { "${params.outdir}/prokka" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: PROKKAGFF2TSV {
+        publishDir = [
+            path: { "${params.outdir}/summary_tables" },
+            mode: params.publish_dir_mode,
+            pattern: "*.prokka-annotations.tsv.gz"
+        ]
+    }
+
+    withName: '.*:FEATURECOUNTS_CDS' {
+        ext.args = '-g ID -t CDS -F gtf'
+        publishDir = [
+            path: { "${params.outdir}/featurecounts" },
+            mode: params.publish_dir_mode,
+            pattern: "*.featureCounts.*"
+        ]
+    }
+
+    withName: COLLECT_FEATURECOUNTS {
+        publishDir = [
+            path: { "${params.outdir}/summary_tables" },
+            mode: params.publish_dir_mode,
+            pattern: "*.gz"
+        ]
+    }
+
+    withName: 'HMMRANK' {
+        publishDir = [
+            path: { "${params.outdir}/summary_tables" },
+            mode: params.publish_dir_mode,
+            pattern: "*.tsv.gz"
+        ]
+    }
+
+    withName: EGGNOG_DOWNLOAD {
+        storeDir = "${params.eggnog_dbpath}"
+    }
+
+    withName: EGGNOG_MAPPER {
+        publishDir = [
+            [
+                path: { "${params.outdir}/eggnog" },
+                mode: params.publish_dir_mode,
+                pattern: "*.emapper.*"
+            ],
+            [
+                path: { "${params.outdir}/summary_tables" },
+                mode: params.publish_dir_mode,
+                pattern: "*.emapper.tsv.gz"
+            ]
+        ]
+    }
+
+    withName: KOFAMSCAN_DOWNLOAD {
+        storeDir = "${params.kofam_dir}"
```

```suggestion
        storeDir = { "${params.kofam_dir}" }
```
Same question here. Is a closure a valid assignment?